### PR TITLE
restructures pod framework to recent advances

### PR DESCRIPTION
**Pull request by xh3b4sd**
This is basically boyscouting and preparations for further efforts towards the update story on KVM.

### CODE REVIEW

File: service/pod/framework.go [hunk 1]
```diff
@@ -11,76 +11,88 @@ import (
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/giantswarm/kvm-operator/service/pod/v2"
-	"github.com/giantswarm/kvm-operator/service/pod/v2/key"
+)
+
+const (
+	PodWatcherLabel = "giantswarm.io/pod-watcher"
```
Comment (xh3b4sd) [inline service/pod/framework.go]: This is some ugly shit I do not have a nice solution for yet. Feedback welcome.

File: service/pod/v2/resource/pod/create.go [hunk 2]
```diff
@@ -4,6 +4,6 @@ import (
 	"context"
 )
 
-func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```
Comment (xh3b4sd) [inline service/pod/v2/resource/pod/create.go]: Going with the interface of the native resource here. Maybe I go back to the CRUD one later but for now should be good enough. Lets see.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

**Review by MarcelMue — APPROVED** [review]
Could add godoc as in AWS-Operator, I am fine with this though <3

**Pull request merged**